R/methods.R

```diff
@@ -1,17 +1,85 @@
+#' to be used in the imputation model. Must be one of `"us"` (default), `"toep"`, `"cs"` or `"ar1"`.
+#' See details.
+#'
+#' @param threshold a numeric between 0 and 1, specifies the proportion of bootstrap
+#' datasets that can fail to produce valid samples before an error is thrown.
+#' See details.
+#'
+#' @param REML a logical indicating whether to use REML estimation rather than maximum
+#' likelihood.
+#'
+#' @param type a character string that specifies the method to use when calculating
+#' confidence intervals. Must be one of `"bootstrap"` (default) or `"percentile"`
+#'
+#' @param verbose a logical, if true (default) Stan's sampling log information will
+#' be printed to the console, if false this information will be suppressed.
+#'
+#' @param seed a numeric that specifies a seed to be used in the call to Stan. This
+#' argument is forward on the the `seed` argument of [rstan::sampling()]. Note that
+#' this is only required for `method_bayes()`, for all other methods you can achieve
+#' reproducible results by setting the seed via `set.seed()`
+#'
+#' @details
+#'
+#' In the case of `method_condmean(type = "bootstrap")` there will be `n_samples + 1`
+#' imputation models and datasets generated as the first sample will be based on
+#' the original dataset whilst the other `n_samples` samples will be
+#' bootstrapped datasets. Likewise, for `method_condmean(type = "jackknife")` there will
+#' be `nrow(data) + 1` imputation models and datasets generated. In both cases this is
+#' represented by `x + 1` being displayed in the print message.
+#'
+#' The user is able to specify different covariance structures using the the `covariance`
+#' argument. Currently supported structures include:
+#'
+#' - Unstructured (`"us"`)
+#' - Toeplitz (`"toep"`)
+#' - Compound Symmetry (`"cs"`)
+#' - Autoregression-1 (`"ar1"`)
+#'
+#' Note that at present Bayesian methods only support unstructured.
+#'
+#' In the case of `method_condmean(type = "bootstrap")` and `method_approxbayes()`, repeated
+#' bootstrap samples of the original dataset are taken with an MMRM fitted to each sample.
+#' Due to the randomness of these sampled datasets, as well as limitations in the optimisers
+#' used to fit the models, it is not uncommon that estimates for a particular dataset can't
+#' be generated. In these instances rbmi is designed to throw out that bootstrapped data
+#' and try again with another. However to ensure that these errors are due to chance and
+#' not due to some underlying misspecification in the data and/or model a tolerance limit
+#' is set on how many samples can be discarded. Once the tolerance limit has been breached
+#' an error will be thrown and the process aborted. The tolerance limit is defined as
+#' `ceiling(threshold * n_samples)`. Note that for jackknife methods estimates need to be
+#' generated for all permutations of dataset and as such an error will be thrown if
```

Comment:
maybe "for all leave-one-out datasets" is better?

Reply:
done